File: patch_dashboard.php
<?php
$filepath = __DIR__ . '/resources/js/Components/Dashboard.jsx';
$content = file_get_contents($filepath);

// 1. Update component props to include selectedSite
$content = preg_replace(
    '/(selectedFundCenter)(\s*}\s*=>\s*\{)/',
    '$1, selectedSite$2',
    $content
);

// 2. Update params objects to include site
$content = preg_replace(
    '/(fund_center:\s*selectedFundCenter,?)/',
    "$1\n                        site: selectedSite,",
    $content
);

// 3. Update useEffect dependencies
$content = preg_replace(
    '/(selectedFundCenter)(\s*\])/',
    '$1, selectedSite$2',
    $content
);

// 4. Add the new state variables
$old_state = "const [commitmentItems, setCommitmentItems] = useState([]);";
$new_state = "const [commitmentItems, setCommitmentItems] = useState([]);\n    const [sites, setSites] = useState([]);\n    const [selectedSite, setSelectedSite] = useState('');";
$content = str_replace($old_state, $new_state, $content);

// 4b. Add API call for sites
$old_fetch_fc = "            try {\n                const fcResponse = await axios.get(route('api.dashboard.fund_centers'));";
$new_fetch_fc = "            try {\n                const sResponse = await axios.get(route('api.dashboard.sites'));\n                setSites(sResponse.data);\n            } catch (err) {\n                console.error(\"Failed to load sites\", err);\n            }\n            try {\n                const fcResponse = await axios.get(route('api.dashboard.fund_centers'));";
$content = str_replace($old_fetch_fc, $new_fetch_fc, $content);

// 5. Add the Site SearchableSelect
$old_filter = "                            {/* Fund Center Filter */}\n                            <SearchableSelect\n                                items={fundCenters}";
$new_filter = "                            {/* Site Filter */}\n                            <SearchableSelect\n                                items={sites}\n                                value={selectedSite}\n                                onChange={(val) => {\n                                    setSelectedSite(val);\n                                    if (val !== selectedSite) setSelectedFundCenter('');\n                                }}\n                                placeholder=\"-- Pilih Site --\"\n                                valueKey=\"Site\"\n                                labelKey=\"Site\"\n                                minWidth=\"100px\"\n                                maxWidth=\"150px\"\n                                displayLabelOnly={true}\n                            />\n\n                            {/* Fund Center Filter */}\n                            <SearchableSelect\n                                items={selectedSite ? fundCenters.filter(fc => fc.Site === selectedSite) : fundCenters}";
$content = str_replace($old_filter, $new_filter, $content);

// 6. Make sure all component usages get selectedSite={selectedSite}
$content = preg_replace(
    '/(selectedFundCenter={selectedFundCenter})/',
    '$1 selectedSite={selectedSite}',
    $content
);

file_put_contents($filepath, $content);
echo "Dashboard.jsx updated successfully!\n";

File: patch_dashboard.py
import re

filepath = r'c:\Users\prayo\.gemini\antigravity-ide\scratch\plo-dashboard-monitoring\resources\js\Components\Dashboard.jsx'

with open(filepath, 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Update component props to include selectedSite
# Matches things like `({ filterType, ..., selectedFundCenter }) => {`
content = re.sub(
    r'(selectedFundCenter)(\s*}\s*=>\s*\{)',
    r'\1, selectedSite\2',
    content
)

# 2. Update params objects to include site
# Matches `fund_center: selectedFundCenter,`
content = re.sub(
    r'(fund_center:\s*selectedFundCenter,?)',
    r'\1\n                        site: selectedSite,',
    content
)

# 3. Update useEffect dependencies
# Matches `selectedFundCenter]`
content = re.sub(
    r'(selectedFundCenter)(\s*\])',
    r'\1, selectedSite\2',
    content
)

# 4. Add the new state variables and API calls to Dashboard component
# Find the start of Dashboard component
dashboard_start = content.find("const [commitmentItems, setCommitmentItems] = useState([]);")
if dashboard_start != -1:
    new_state = """const [commitmentItems, setCommitmentItems] = useState([]);
    const [sites, setSites] = useState([]);
    const [selectedSite, setSelectedSite] = useState('');"""
    content = content.replace("const [commitmentItems, setCommitmentItems] = useState([]);", new_state)

fetch_items_start = content.find("const sResponse = await axios.get(route('api.dashboard.sites'));")
if fetch_items_start == -1: # if not already added
    old_fetch_fc = """            try {
                const fcResponse = await axios.get(route('api.dashboard.fund_centers'));"""
    new_fetch_fc = """            try {
                const sResponse = await axios.get(route('api.dashboard.sites'));
                setSites(sResponse.data);
            } catch (err) {
                console.error("Failed to load sites", err);
            }
            try {
                const fcResponse = await axios.get(route('api.dashboard.fund_centers'));"""
    content = content.replace(old_fetch_fc, new_fetch_fc)

# 5. Add the Site SearchableSelect
old_filter = """                            {/* Fund Center Filter */}
                            <SearchableSelect
                                items={fundCenters}"""
new_filter = """                            {/* Site Filter */}
                            <SearchableSelect
                                items={sites}
                                value={selectedSite}
                                onChange={(val) => {
                                    setSelectedSite(val);
                                    if (val !== selectedSite) setSelectedFundCenter('');
                                }}
                                placeholder="-- Pilih Site --"
                                valueKey="Site"
                                labelKey="Site"
                                minWidth="100px"
                                maxWidth="150px"
                                displayLabelOnly={true}
                            />

                            {/* Fund Center Filter */}
                            <SearchableSelect
                                items={selectedSite ? fundCenters.filter(fc => fc.Site === selectedSite) : fundCenters}"""
content = content.replace(old_filter, new_filter)

# 6. Make sure all component usages get selectedSite={selectedSite}
# E.g. <MonthlyTrendChart ... selectedFundCenter={selectedFundCenter} />
content = re.sub(
    r'(selectedFundCenter={selectedFundCenter})',
    r'\1 selectedSite={selectedSite}',
    content
)

with open(filepath, 'w', encoding='utf-8') as f:
    f.write(content)

print("Dashboard.jsx updated successfully!")
